rllib/examples/env/multi_agent.py

```diff
@@ -87,6 +88,62 @@ def step(self, action_dict):
         return obs, rew, done, info
 
 
+class FlexAgentsMultiAgent(MultiAgentEnv):
+    """Env of independent agents, each of which exits after n steps."""
+
+    def __init__(self):
+        self.agents = {}
+        self.agentID = 0
+        self.dones = set()
+        self.observation_space = gym.spaces.Discrete(2)
+        self.action_space = gym.spaces.Discrete(2)
+        self.resetted = False
+
+    def spawn(self):
+        # Spawn a new agent into the current episode.
+        agentID = self.agentID
+        self.agents[agentID] = MockEnv(25)
+        self.agentID += 1
+        return agentID
+
+    def reset(self):
+        self.agents = {}
+        self.spawn()
+        self.resetted = True
+        self.dones = set()
+
+        obs = {}
+        for i, a in self.agents.items():
+            obs[i] = a.reset()
+
+        return obs
+
+    def step(self, action_dict):
+        obs, rew, done, info = {}, {}, {}, {}
+        # Apply the actions.
+        for i, action in action_dict.items():
+            obs[i], rew[i], done[i], info[i] = self.agents[i].step(action)
+            if done[i]:
+                self.dones.add(i)
+
+        # Sometimes, add a new agent to the episode.
+        if random.random() > 0.75:
```

Comment:
What is the motivation behind killing/creating new agents during the episode?

Reply:
Applicability to an entire new class of environments, including artificial life inspired work. I'm the author of one such project in which this bug broke the trajectory API https://jsuarez5341.github.io/neural-mmo/build/html/rst/userguide.html

Reply:
@michaelzhiluo This is just an example env to simulate this kind of behavior in real envs.

Reply:
Very cool! Aight approved.